### PR TITLE
Add cffi package

### PR DESCRIPTION
**Pull request by hoodmane**

### CODE REVIEW

**Comment by hoodmane** [comment]
@rth I think this works now (tests pass locally).

File: pyodide-build/pyodide_build/buildpkg.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def download_and_extract(
         srcdir = Path(pkg["source"]["path"])
 
         if not srcdir.is_dir():
-            raise ValueError("'path' must point to a path")
+            raise ValueError("'path' must point to a directory that exists")
```
Comment (rth) [inline pyodide-build/pyodide_build/buildpkg.py:102]: Maybe,
```suggestion
            raise ValueError(f"path={srcdir} must point to a directory that exists")
```
while we are at it,

File: packages/cffi_example/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  name: cffi_example
+  version: "0.1"
+requirements:
+  run:
+    - cffi
+source:
+  path: python-cffi-example
+test:
+  imports:
+    - cffi_example
```
Comment (rth) [inline packages/cffi_example/meta.yaml:11]: Could you add about/home link pointing to github as well and the license? Actually can't we install it from a Github release? Or did you make lots of changes to the example?

**Review by rth — APPROVED** [review]
Great! A few minor comments otherwise LGTM.

File: packages/cffi_example/meta.yaml (hunk 2 above)
Comment (hoodmane) [reply packages/cffi_example/meta.yaml:11]: I did not make many any changes to the example (just deleted some unneeded files). I don't think there is support in `pyodide-build` for loading a github repo, and they don't have seem to have made any releases. But I agree we should fix it to download it from github in some way.

Comment (rth) [reply packages/cffi_example/meta.yaml:11]: Even without a release you should be able to download it as https://github.com/wolever/python-cffi-example/archive/refs/heads/master.zip  Or with the appropriate commit in the URL. It's the "Download Zip" button,
image

**Pull request merged**